woocommerce/woocommerce
# Update the select all behavior in the variations table

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #40214, #41246



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure feature `product-variation-management` is enabled under `Features` tab from `/wp-admin/tools.php?page=woocommerce-admin-test-helper` (WooCommerce Beta Tester plugin) is required
3. Go to `/wp-admin/admin.php?page=wc-admin&path=/add-product&tab=variations` and add some attributes to generate variations
4. Wait until variations get generated
5. When selecting at least one variation some selection actions appear at the header of the table. 
<img width="675" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/7f3b60f8-c683-42b4-934e-047d156fd4a0">

6. The first element should show the amount of selected variations like `N selected`
7. Clicking `Select page (N)` should select all the items present in the current page, where `N` should indicate that number.
8. `Select all (N)` should select all the items across all the existing pages in the table.
9. `Clear selection` should remove the selection
10. If at least one variation is selected `Edit` button and the three dots quick actions menu per row should disappear
11. When all items are selected, the selection persists when the user changes pages. So, if I select 5 items on page 2 and 3 items on page 2, I'll see the selected items while switching between pages.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
